tools/gengen/util/generator.go

```diff
@@ -567,9 +510,11 @@ func (g *GenesisGenerator) getDealWeight(dealID abi.DealID, endEpoch abi.ChainEp
 	if err != nil {
 		return specsbig.Zero(), err
 	}
-	return *weightOut.(*specsbig.Int), nil
+	ret := weightOut.(*market.VerifyDealsOnSectorProveCommitReturn)
+	return ret.VerifiedDealWeight, nil
 }
 
+// TODO: This might not be right
```

Comment:
Yes from testing powertable view with gen gen state I'm confident it is not right.  Could you please file an issue for measuring power properly?  Something else you can include in the issue: I'm also worried that our IsWinner functionality is reading the wrong notion of power (right now it just uses sectors * sectorsize and rawbytenetwork power so I believe that either needs to change now or soon too.

Reply:
#4011